tox.ini

```diff
@@ -29,7 +29,7 @@ skip_install = true
 commands =
   black --check .
   flake8 monocle
-  mypy --ignore-missing-imports monocle
+  mypy --ignore-missing-imports --python-version 3.9 monocle
```

Comment:
If the plan is to make a new GHA specific to mypy then I prefer we avoid the Makefile option and instead split the tox  linters target in two new targets: format and mypy.

Reply:
Breaking tox to static_anaylisys, unit, functional can also work. The reason behind it is to make the CI quicker. All workflows are run in parallel.

Reply:
> If the plan is to make a new GHA specific to mypy then I prefer we avoid the Makefile option and instead split the tox linters target in two new targets: format and mypy.

It is not a mypy specific workflow. We can split tox as you suggested.